zetaclient/config/types.go

```diff
@@ -17,93 +21,256 @@ type ClientConfiguration struct {
 	SignerPasswd    string
 }
 
-type CoreParams struct {
-	ChainID                     int64
-	GasPriceTicker              uint64
-	InTxTicker                  uint64
-	OutTxTicker                 uint64
-	WatchUTXOTicker             uint64
-	ConfCount                   uint64
-	ConnectorContractAddress    string
-	ZETATokenContractAddress    string
-	ERC20CustodyContractAddress string
-	OutboundTxScheduleInterval  int64
-	OutboundTxScheduleLookahead int64
-}
-
-func NewCoreParams() *CoreParams {
-	return &CoreParams{
-		ChainID:                     0,
-		GasPriceTicker:              0,
-		InTxTicker:                  5,
-		OutTxTicker:                 3,
-		WatchUTXOTicker:             5,
-		ConfCount:                   0,
-		ConnectorContractAddress:    "",
-		ZETATokenContractAddress:    "",
-		ERC20CustodyContractAddress: "",
-	}
-}
-
 type EVMConfig struct {
-	Client     *ethclient.Client
-	Chain      common.Chain
-	Endpoint   string
-	CoreParams *CoreParams
+	observertypes.CoreParams
+	Chain    common.Chain
+	Endpoint string
 }
 
 type BTCConfig struct {
+	observertypes.CoreParams
+
 	// the following are rpcclient ConnConfig fields
 	RPCUsername string
 	RPCPassword string
 	RPCHost     string
 	RPCParams   string // "regtest", "mainnet", "testnet3"
-
-	CoreParams *CoreParams
 }
 
+// TODO: use snake case for json fields
 type Config struct {
-	Peer          string
-	PublicIP      string
-	LogFormat     string
-	LogLevel      zerolog.Level
-	LogSampler    bool
-	PreParamsPath string
-	Keygen        zetaObserverTypes.Keygen
-	ChainID       string
-	ZetaCoreURL   string
-	AuthzGranter  string
-	AuthzHotkey   string
-
-	ChainsEnabled       []common.Chain
-	EVMChainConfigs     map[int64]*EVMConfig // TODO : chain to chain id
-	BitcoinConfig       *BTCConfig
-	P2PDiagnostic       bool
-	ConfigUpdateTicker  uint64
-	P2PDiagnosticTicker uint64
-	TssPath             string
-	TestTssKeysign      bool
+	Peer                string        `json:"Peer"`
```

Comment:
Maybe we can create an issue on Github for changing config JSON fields to snake_case?